src/osx/cocoa/button.mm

```diff
@@ -187,9 +187,18 @@ - (NSControlSize)controlSize;
     {
         wxString accelstring(label[accelPos + 1]); // Skip '&' itself
         accelstring.MakeLower();
-        wxCFStringRef cfText(accelstring);
-        [GetNSButton() setKeyEquivalent:cfText.AsNSString()];
-        [GetNSButton() setKeyEquivalentModifierMask:NSCommandKeyMask];
+        // Avoid Cmd+C closing dialog on Mac.
+        if (accelstring == "c" && GetWXPeer()->GetId() == wxID_CANCEL)
+        {
+            [GetNSButton() setKeyEquivalent:@""];
```

Comment:
Can we use something that would make Escape be the key equivalent for the button? Or does this already work due to something else?

Reply:
Escape already works due to something else.